Sources/LibMobileCoin/McRngCallback.swift

```diff
@@ -5,15 +5,34 @@
 import Foundation
 import LibMobileCoin
 
+func withMcRngObjCallback<T>(
+    rngFunc: (@convention(c) (UnsafeMutableRawPointer?) -> UInt64)?,
+    rng: MobileCoinRng,
+    _ body: (UnsafeMutablePointer<McRngCallback>?) throws -> T
+) rethrows -> T {
+    if let rngFunc = rngFunc {
+        let rawRng = Unmanaged.passUnretained(rng).toOpaque()
+        var rngCallback = McRngCallback(rng: rngFunc, context: rawRng)
+        return try body(&rngCallback)
+    } else {
+        return try body(nil)
+    }
+}
+
 func withMcRngCallback<T>(
     rng: (@convention(c) (UnsafeMutableRawPointer?) -> UInt64)?,
     rngContext: Any?,
     _ body: (UnsafeMutablePointer<McRngCallback>?) throws -> T
 ) rethrows -> T {
     if let rng = rng {
-        var rngContext = rngContext
-        return try withUnsafeMutablePointer(to: &rngContext) { rngContextPtr in
-            var rngCallback = McRngCallback(rng: rng, context: rngContextPtr)
+        if let rngContext = rngContext {
```

Comment:
this change isn't needed, the existing rust code defaults to `McRng` if the callback ptr is nil.

```rust
impl<'a, 'b> FromFfi<FfiOptMutPtr<'a, McRngCallback<'b>>> for SdkRng<'a, 'b> {
    #[inline]
    fn from_ffi(src: FfiOptMutPtr<'a, McRngCallback<'b>>) -> Self {
        if let Some(callback) = src.into_mut() {
            SdkRng::CallbackRng(CallbackRng(callback.rng.0, &mut callback.context))
        } else {
            SdkRng::McRng(McRng::default())
        }
    }
}
```

Reply:
Right... I originally started out sharing the same `func withMcRngCallback<T>(...)` func when it seemed like 'context' wasn't being used anywhere (I didn't notice the test code).  I should have removed this when I switched to a different func to explicitly handle passing in MobileCoinRng.

Code updated.